certora/README.md

```diff
@@ -0,0 +1,60 @@
+This folder contains the verification of the universal rewards distribution mechanism using CVL, Certora's Verification Language.
+
+# Folder and file structure
+
+The [`certora/specs`](specs) folder contains the specification files.
+
+The [`certora/confs`](confs) folder contains a configuration file for each corresponding specification file.
+
+The [`certora/helpers`](helpers) folder contains files that enable the verification of Morpho Blue.
+
+# Overview of the verification
+
+This work aims at providing a formally verified rewards checker.
+The rewards checker is composed of the [Checker.sol](checker/Checker.sol) file, which takes a certificate as an input.
+The certificate is assumed to contain the submitted root to verify, a total amount of rewards distributed, and a Merkle tree, and it is checked that:
+
+1. the Merkle tree is a well-formed Merkle tree
+2. the total amount of rewards distributed matches the total rewards contained in the Merkle tree
+
+Those checks are done by only using "trusted" functions, namely `newLeaf` and `newInternalNode`, that have been formally verified to preserve those invariants:
```

Comment:
The checks are done using the wellFormed functions no ?

Reply:
The rewards checker just calls `newLeaf` and `newInternalNode`, it does not call well formed functions. Those 2 functions have `require` statement at the beginning, so the checker is doing those checks, and this is the reason why the `wellFormed` invariant holds.
I reworded to remove the total amounts which were artifacts from the morpho-optimizers rewards checker

Reply:
okok, just to make sure I have it completely:

newLeaf and newInternalNode can only create well formed trees or revert
and this we verify it (with wellformed functions)
so we don't even need to check that a tree is well formed once it has been formed by these functions